test/integration_test/kubevirt_hyperconv_test.go

```diff
@@ -280,6 +280,64 @@ func kubeVirtHypercVPSFixJob(t *testing.T) {
 	log.Infof("Test status at end of %s test: %s", t.Name(), testResult)
 }
 
+// This test simulates OCP upgrade by live-migrating all VMs from a node a NON-replica node and
+// then restarting PX on the node where volume is attached.
+// It expects that all the VMs from that node should end up with a bind-mount (hyperconvergence).
+// PX performs a single live migration for each VM on the node.
+// The above steps are repeated for all worker nodes in the cluster
+func kubeVirtSimulateOCPUpgrade(t *testing.T) {
+	var testrailID, testResult = 297265, testResultFail
+	runID := testrailSetupForTest(testrailID, &testResult)
+	defer updateTestRail(&testResult, testrailID, runID)
+	instanceID := "per-node"
+
+	ctxs := kubevirtVMScaledDeployAndValidate(
+		t,
+		instanceID,
+		[]string{
+			"kubevirt-fedora", "kubevirt-fedora-wait-first-consumer", "kubevirt-fedora-multi-disks-wffc",
+			"kubevirt-windows-22k-server", "kubevirt-windows-22k-server-wait-first-consumer",
+		},
+		kubevirtScale,
+	)
+	allNodes := node.GetNodesByVoDriverNodeID()
+
+	nodeTestStateMap := getNodeVMTestStateMap(t, ctxs, allNodes)
+
+	for currNode, testStatesNode := range nodeTestStateMap {
+		log.Infof("\nStart upgrade simulation on node: %s", currNode.Name)
```

Comment:
instead of creating the nodeTestStateMap beforehand, get the testState for all VMs currently on this node. Note that the list of VMs on each node is going to change as the other nodes are upgraded since some of the VMs on the node being upgraded would have moved to this node.

Reply:
Fixed.